osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -89,6 +89,12 @@ private void load()
             };
         }
 
+        public void ShowWithSearch(string query)
+        {
+            filterControl.Search(query);
+            Show();
+        }
```

Comment:
Currently performing a search from this method does not reset the current states of the filter control, this does not match osu!web's behaviour and sounds wrong in general.

Reply:
I've looked into it and it's not as straightforward as you would imagine, for several reasons:

* No robust way to iterate over all bindables in the search control (you have to manually `.Clear()`/`.SetDefault()` them one by one)
* The former is also blocked by `IBindableList` not implementing `IHasDefault`
* `BeatmapSearchMultipleSelectionFilterRow` is written in a one-way manner (UI interactions change `Current`, but `Current` changes do not change UI state).

So I'd leave as a follow-up/prerequisite.